element-hq/element-x-android
# Use formatted captions for images and video

## Content

This PR has the following changes:

- Parse the `caption` and `formattedCaption` in the same way as we do for text messages.
- Allow opening links from the caption UI text component.
- Split the `onClick` action for timeline items between those whose onClick action should be triggered by the whole message bubble and those who should trigger it only from certain UI elements (as the image/video preview in the timeline).

Note there's no way to send captions as either markdown or HTML at the moment from the app, so either you send these from some other client or you'd only be able to test this by sending a URL as part of the plain text caption.

This PR can be reviewed commit-by-commit.

## Motivation and context

Fixes #3850.

## Screenshots / GIFs

No changes in screenshots, I hope.

## Tests



- Send an image/video with an URL in the caption and some other text.
- Make sure the link in the caption is later clickable in the timeline item.
- Clicking on the image/video preview should open the viewer.
- Clicking on the text part that has no URL shouldn't do anything.
- Clicking on the URL should open it.
- Clicking on any other timeline items should have the same effect as before this patch.

## Tested devices

- OS version(s): 14

## Checklist
